framework/src/executioners/FEProblemSolve.C

```diff
@@ -254,60 +273,179 @@ FEProblemSolve::FEProblemSolve(Executioner & ex)
+              paramError("ignore_variables_for_autoscaling",
+                         "Variables cannot be in a scaling grouping and also be ignored");
+      }
+      nl.ignoreVariablesForAutoscaling(
+          getParamFromNonlinearSystemVectorParam<std::vector<std::string>>(
+              "ignore_variables_for_autoscaling", i_nl_sys));
     }
-    _nl.ignoreVariablesForAutoscaling(
-        getParam<std::vector<std::string>>("ignore_variables_for_autoscaling"));
   }
 
+  // Multi-grid options
   _problem.numGridSteps(_num_grid_steps);
 }
 
+template <typename T>
+T
+FEProblemSolve::getParamFromNonlinearSystemVectorParam(std::string param_name,
+                                                       unsigned int index) const
+{
+  const auto & param_vec = getParam<std::vector<T>>(param_name);
+  if (index > _num_nl_systems)
+    paramError(param_name,
+               "Vector parameter is requested at index (" + std::to_string(index) +
+                   ") which is larger than number of nonlinear systems (" +
+                   std::to_string(_num_nl_systems) + ").");
+  if (param_vec.size() == 0)
+    paramError(
+        param_name,
+        "This parameter was passed to a routine which cannot handle empty vector parameters");
+  if (param_vec.size() != 1 && param_vec.size() != _num_nl_systems)
+    paramError(param_name,
+               "Vector parameter size (" + std::to_string(param_vec.size()) +
+                   ") is different than the number of nonlinear systems (" +
+                   std::to_string(_num_nl_systems) + ").");
+
+  // User passed only one parameter, assume it applies to all nonlinear systems
+  if (param_vec.size() == 1)
+    return param_vec[0];
+  else
+    return param_vec[index];
+}
+
 bool
 FEProblemSolve::solve()
 {
-  // This loop is for nonlinear multigrids (developed by Alex)
+  // This should be late enough to retrieve the convergence object.
+  // TODO: Move this to a setup phase, which does not exist for SolveObjects
+  if (isParamValid("multi_system_fixed_point_convergence"))
+    _multi_sys_fp_convergence =
+        &_problem.getConvergence(getParam<ConvergenceName>("multi_system_fixed_point_convergence"));
+
+  // Outer loop for multi-grid convergence
+  bool converged = false;
+  unsigned int num_fp_multisys_iters = 0;
   for (MooseIndex(_num_grid_steps) grid_step = 0; grid_step <= _num_grid_steps; ++grid_step)
   {
-    _problem.solve(_nl.number());
-
-    if (_problem.shouldSolve())
+    // Multi-system fixed point loop
+    // Use a convergence object if provided, if not, use a reasonable default of every nested system
+    // being converged
+    num_fp_multisys_iters = 0;
+    converged = false;
+    while (_multi_sys_fp_convergence
+               ? (_multi_sys_fp_convergence->checkConvergence(num_fp_multisys_iters) ==
+                  Convergence::MooseConvergenceStatus::ITERATING)
+               : !converged)
```

Comment:
If you don't have the convergence object and have multisystem could you iterate forever? I think that if you have multisystem, the convergence object should be a required parameter

Reply:
See the constructor, it is required

Reply:
Maybe just add a `_have_multiple_sys` boolean then that is constructed from `_multi_sys_fp_convergence`? I think it would slightly increase readability to help dummies like me who might not read all the code better grok what's going on

Reply:
Or do something to remove this ternary? I just think this statement and the logic at the end of the loop are hard to read. It would be nice if this was just like `while (!converged)`

Reply:
Or `while (iterating)`

Reply:
will do. Got a few commitments today so it will have to wait for tomorrow.
I m planning a follow up once https://github.com/libMesh/libmesh/pull/4007 is merged if you are ok with pushing that change until then

Reply:
sure